src/simulation/dynamics/constraintEffector/constraintDynamicEffector.cpp

```diff
@@ -198,25 +233,94 @@ void ConstraintDynamicEffector::computeForceTorque(double integTime, double time
             Eigen::Matrix3d dcm_B1B2 = dcm_B1N * dcm_B2N.transpose();
             Eigen::Vector3d L_B2_att = -this->k_a * eigenMRPd2Vector3d(sigma_B2B1) - this->c_a * 0.25 * sigma_B2B1.Bmat() * omega_B2B1_B2;
             Eigen::Vector3d L_B1_att = - dcm_B1B2 * L_B2_att;
-            this->L_B2 = L_B2_len + L_B2_att; // store the constraint torque for spacecraft 2
+            this->T_B2 = L_B2_len + L_B2_att; // store the constraint torque for spacecraft 2
 
             // assign forces and torques for spacecraft 1
             this->forceExternal_N = this->Fc_N;
             this->torqueExternalPntB_B = L_B1_len + L_B1_att;
+            this->T_B1 = this->torqueExternalPntB_B;       
         }
         else if (this->scID == 1) {
             // assign forces and torques for spacecraft 2
             this->forceExternal_N = - this->Fc_N;
-            this->torqueExternalPntB_B = this->L_B2;
+            this->torqueExternalPntB_B = this->T_B2;
         }
         this->scID = (1 + pow(-1,this->scID))/2; // toggle spacecraft to be assigned forces and torques
     }
 }
 
-/*! Update state method, nothing to report here
+/*! This method takes the computed constraint force and torque states and outputs them to the m
+ messaging system.
+ @return void
+ @param CurrentClock The current simulation time (used for time stamping)
+ */
+void ConstraintDynamicEffector::writeOutputStateMessage(uint64_t CurrentClock)
+{
+    ConstDynEffectorMsgPayload outputForces;
+    outputForces = this->constraintElements.zeroMsgPayload;
+    eigenVector3d2CArray(this->forceExternal_N,outputForces.Fc_N);
+    eigenVector3d2CArray(this->T_B1,outputForces.L_B1);
+    eigenVector3d2CArray(this->T_B2,outputForces.L_B2);
+    eigenVector3d2CArray(this->psi_N,outputForces.psi_N);
+    outputForces.F_filtered = this->F_filtered_mag_t;
+    outputForces.T1_filtered = this->T1_filtered_mag_t;
+    outputForces.T2_filtered = this->T2_filtered_mag_t;
+    this->constraintElements.write(&outputForces,this->moduleID,CurrentClock);
+}
+
+/*! Update state method
  @return void
+ @param CurrentSimNanos current simulation time
  */
 void ConstraintDynamicEffector::UpdateState(uint64_t CurrentSimNanos)
+{
+    this->readInputMessage();
+    if(this->effectorStatus){
+    this->computeFilteredForce(CurrentSimNanos);
```

Comment:
Inside the {} brackets indent the code by 4 spaces

Reply:
Resolved.